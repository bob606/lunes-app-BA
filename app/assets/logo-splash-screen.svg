<svg xmlns="http://www.w3.org/2000/svg" width="124" height="100" viewBox="0 0 124 100">
    <defs>
        <filter id="ezx74xqexa">
            <feColorMatrix in="SourceGraphic" values="0 0 0 0 0.976471 0 0 0 0 0.980392 0 0 0 0 0.984314 0 0 0 1.000000 0"/>
        </filter>
    </defs>
    <g fill="none" fill-rule="evenodd">
        <g>
            <g>
                <g>
                    <g>
                        <path fill="#97AEF1" d="M49.536 15.037c4.122 0 7.464-3.346 7.464-7.474S53.658.088 49.536.088c-4.123 0-7.465 3.347-7.465 7.475s3.342 7.474 7.465 7.474M28.997 51.73c-3.272 0-5.926-2.658-5.926-5.936s2.654-5.935 5.926-5.935c7.438 0 13.823-5.552 14.852-12.915.454-3.245 3.45-5.509 6.69-5.054 3.24.454 5.5 3.454 5.046 6.701C53.74 41.781 42.31 51.73 28.997 51.73" transform="translate(-98 -234) translate(98.986 234) translate(0 1.27) translate(34)"/>
                        <path fill="#425BEE" d="M28.011 51.73c-13.516 0-24.956-10.021-26.609-23.31-.4-3.217 1.905-6.146 5.148-6.543 3.247-.399 6.198 1.889 6.598 5.106.922 7.416 7.312 13.01 14.863 13.01 3.268 0 5.918 2.627 5.918 5.868 0 3.241-2.65 5.868-5.918 5.868M7.5 15c4.142 0 7.5-3.358 7.5-7.5C15 3.358 11.642 0 7.5 0 3.358 0 0 3.358 0 7.5 0 11.642 3.358 15 7.5 15" transform="translate(-98 -234) translate(98.986 234) translate(0 1.27) translate(34)"/>
                    </g>
                    <g filter="url(#ezx74xqexa)" transform="translate(-98 -234) translate(98.986 234) translate(0 1.27)">
                        <g>
                            <path fill="#F9FAFB" d="M21.651 1.364l7.26.013-.001.137v.075l-.001.45v.484l-.001.187v.408c-.006 2.585-.014 7.552-.019 11.442v.637l-.001.31v.602l-.001.29v.822l-.001.256v1.327c0 2.188 1.897 3.969 4.23 3.969 2.332 0 4.23-1.781 4.23-3.97V1.37h7.258v17.434C44.604 24.978 39.451 30 33.116 30s-11.49-5.022-11.49-11.196v-1.328l.001-.256v-.823l.001-.29v-.602l.001-.31v-.639l.019-11.45v-.409l.001-.186.001-.484v-.135l.001-.391v-.137zm91.86 0c1.514 0 3.048.297 4.603.892 1.554.595 2.94 1.53 4.156 2.804l-3.487 4.588c-.892-.65-1.75-1.147-2.575-1.487-.824-.34-1.723-.51-2.697-.51-.865 0-1.54.157-2.027.468-.486.312-.73.78-.73 1.402 0 .652.345 1.155 1.034 1.508.69.355 1.643.772 2.859 1.254l3.203 1.317c1.649.68 2.92 1.629 3.812 2.846.892 1.219 1.338 2.79 1.338 4.717 0 1.162-.23 2.28-.69 3.356-.46 1.077-1.128 2.018-2.006 2.825-.88.809-1.954 1.452-3.224 1.934-1.271.482-2.73.722-4.38.722-1.676 0-3.4-.319-5.169-.956-1.771-.638-3.373-1.635-4.806-2.995l3.974-5.013c.946.793 1.967 1.437 3.062 1.932 1.095.497 2.128.744 3.102.744 1.055 0 1.817-.17 2.29-.51.473-.34.711-.82.711-1.445 0-.34-.082-.622-.244-.85-.162-.225-.399-.438-.71-.637-.31-.198-.683-.388-1.115-.573-.433-.184-.932-.39-1.5-.616l-3.285-1.445c-.703-.283-1.378-.658-2.027-1.126-.649-.467-1.216-1.02-1.703-1.656-.487-.637-.872-1.367-1.156-2.189-.284-.821-.426-1.727-.426-2.72 0-1.188.244-2.3.73-3.334.487-1.034 1.162-1.94 2.028-2.72.864-.778 1.899-1.395 3.102-1.848 1.202-.453 2.52-.68 3.953-.68zm-15.25 0v6.042H88.127v4.28h8.663v6.041h-8.663v4.867H98.67v6.042H81.1V1.364H98.26zM61.5 0c6.336 0 11.49 5.022 11.49 11.196v.193l-.001 1.657v.847l-.001.298v.614l-.001.317-.001.648-.015 9.648-.001.53-.002 1.156v.385l-.001.175v.444l-.001.12v.408l-7.26-.013v-.212l.001-.09.001-.51v-.52l.002-.63v-.473c.005-2.465.012-6.384.016-9.755v-.666l.002-.964v-.615l.001-.297v-.846l.001-.266v-1.583c0-2.188-1.897-3.968-4.23-3.968-2.332 0-4.23 1.78-4.23 3.968v17.433h-7.258V11.196C50.011 5.022 55.165 0 61.499 0zM7.23 1.364v21.231h10.341v6.041H0V1.364h7.23z" transform="translate(0 68.73)"/>
                        </g>
                    </g>
                </g>
            </g>
        </g>
    </g>
</svg>
