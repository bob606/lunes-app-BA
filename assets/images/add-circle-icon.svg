<?xml version="1.0" encoding="UTF-8"?>
<svg width="32px" height="31px" viewBox="0 0 32 31" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <title>D09AE073-3E12-4E4C-9490-EA94D5F6FBD9</title>
    <g id="Individual-Areas" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="S-Viewport---Add-Area---V2" transform="translate(-15.000000, -147.000000)">
            <g id="CTA-/-light-/-L-/-back" transform="translate(15.375000, 147.000000)">
                <circle id="outline" stroke="#E0E4ED" fill="#FFFFFF" cx="15.5" cy="15.5" r="15"></circle>
                <g id="Icon-24px-/-add" transform="translate(15.666667, 15.666667) rotate(-360.000000) translate(-15.666667, -15.666667) translate(5.166667, 5.166667)" fill="#000E38">
                    <polygon id="Path" points="16.3611111 11.1944444 11.1944444 11.1944444 11.1944444 16.3611111 9.47222222 16.3611111 9.47222222 11.1944444 4.30555556 11.1944444 4.30555556 9.47222222 9.47222222 9.47222222 9.47222222 4.30555556 11.1944444 4.30555556 11.1944444 9.47222222 16.3611111 9.47222222"></polygon>
                </g>
            </g>
        </g>
    </g>
</svg>