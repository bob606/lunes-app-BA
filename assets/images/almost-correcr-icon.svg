<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
    <defs>
        <filter id="f77tzr9qua">
            <feColorMatrix in="SourceGraphic" values="0 0 0 0 0.227451 0 0 0 0 0.250980 0 0 0 0 0.286275 0 0 0 1.000000 0"/>
        </filter>
    </defs>
        <g>
            <g>
                <g transform="translate(-31 -331) translate(15 308) translate(16 23)">
                    <circle cx="12" cy="12" r="11.5" fill="none"/>
                    <g filter="url(#f77tzr9qua)" transform="translate(4 4)">
                        <g>
                            <path d="M-0.495 5.111L12.655 5.469 13.173 6.334 -1.013 5.976z" transform="translate(1.798 2.19) rotate(-45 6.08 5.723)"/>
                            <path d="M4.858.663L4.194 0 2.578 1.617.961 0 .298.663 1.914 2.28.298 3.897.961 4.56 2.578 2.943 4.194 4.56 4.858 3.897 3.241 2.28zM8.243 10.421L6.757 8.935 6.079 9.609 8.243 11.774 12.458 7.559 11.784 6.886z" transform="translate(1.798 2.19)"/>
                        </g>
                    </g>
                </g>
            </g>
        </g>
</svg>
