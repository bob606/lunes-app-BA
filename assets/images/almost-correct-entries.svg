<svg xmlns="http://www.w3.org/2000/svg" width="36" height="36" viewBox="0 0 36 36">
    <defs>
        <filter id="toeahqusoa">
            <feColorMatrix in="SourceGraphic" values="0 0 0 0 0.000000 0 0 0 0 0.054902 0 0 0 0 0.219608 0 0 0 1.000000 0"/>
        </filter>
    </defs>
    <g fill="none" fill-rule="evenodd">
        <g>
            <g transform="translate(-143 -88) translate(143 88)">
                <circle cx="18" cy="18" r="17.5" stroke="#000E38"/>
                <g filter="url(#toeahqusoa)" transform="translate(6 6)">
                    <g fill="#000">
                        <path d="M-0.743 7.666L18.983 8.204 19.76 9.501 -1.52 8.964z" transform="translate(2.697 3.284) rotate(-45 9.12 8.584)"/>
                        <path d="M7.286.995L6.291 0 3.866 2.425 1.442 0 .446.995 2.871 3.42.446 5.845 1.442 6.84 3.866 4.415 6.291 6.84 7.286 5.845 4.862 3.42zM12.365 15.632L10.136 13.403 9.118 14.414 12.365 17.661 18.686 11.339 17.676 10.328z" transform="translate(2.697 3.284)"/>
                    </g>
                </g>
            </g>
        </g>
    </g>
</svg>
