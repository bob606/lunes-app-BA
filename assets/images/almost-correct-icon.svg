<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
    <defs>
        <filter id="irueecwzka">
            <feColorMatrix in="SourceGraphic" values="0 0 0 0 0.000000 0 0 0 0 0.054902 0 0 0 0 0.219608 0 0 0 1.000000 0"/>
        </filter>
    </defs>
    <g fill="none" fill-rule="evenodd">
        <g>
            <g>
                <g transform="translate(-31 -331) translate(15 308) translate(16 23)">
                    <circle cx="12" cy="12" r="11.5" fill="#FFF" stroke="#E0E4ED"/>
                    <g filter="url(#irueecwzka)" transform="translate(4 4)">
                        <g fill="#000">
                            <path d="M-0.495 5.111L12.655 5.469 13.173 6.334 -1.013 5.976z" transform="translate(1.798 2.19) rotate(-45 6.08 5.723)"/>
                            <path d="M4.858.663L4.194 0 2.578 1.617.961 0 .298.663 1.914 2.28.298 3.897.961 4.56 2.578 2.943 4.194 4.56 4.858 3.897 3.241 2.28zM8.243 10.421L6.757 8.935 6.079 9.609 8.243 11.774 12.458 7.559 11.784 6.886z" transform="translate(1.798 2.19)"/>
                        </g>
                    </g>
                </g>
            </g>
        </g>
    </g>
</svg>
