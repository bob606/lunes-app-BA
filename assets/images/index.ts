import AddCircleIcon from './add-circle-icon.svg'
import ArrowLeftCircleIconBlue from './arrow-left-circle-icon-blue.svg'
import ArrowLeftCircleIconWhite from './arrow-left-circle-icon-white.svg'
import ArrowRightCircleIconWhite from './arrow-right-circle-icon-white.svg'
import ArrowRightIcon from './arrow-right-icon.svg'
import BannerGreen from './banner-green.png'
import BannerRed from './banner-red.png'
import BannerYellow from './banner-yellow.png'
import BookIconGrey from './book-icon-grey.svg'
import BookIconWhite from './book-icon-white.svg'
import CheckCircleIconBold from './check-circle-icon-bold.svg'
import CheckCircleIconGreen from './check-circle-icon-green.svg'
import CheckCircleIconWhite from './check-circle-icon-white.svg'
import CheckCircleIcon from './check-circle-icon.svg'
import CheckCloseCircleIconBold from './check-close-circle-icon-bold.svg'
import CheckCloseCircleIcon from './check-close-circle-icon.svg'
import ChevronRight from './chevron-right.svg'
import CloseCircleIconBlue from './close-circle-icon-blue.svg'
import CloseCircleIconBold from './close-circle-icon-bold.svg'
import CloseCircleIconWhite from './close-circle-icon-white.svg'
import CloseCircleIcon from './close-circle-icon.svg'
import CloseIconRed from './close-icon-red.svg'
import CloseIconWhite from './close-icon-white.svg'
import CloseIcon from './close-icon.svg'
import DoubleCheckCircleIconWhite from './double-check-circle-icon-white.svg'
import DoubleCheckIcon from './double-check-icon.svg'
import HeaderCircleIcon from './header-cirlce-icon.svg'
import HeaderLinesIcon from './header-lines-icon.svg'
import HeaderSquareIcon from './header-square-icon.svg'
import HeaderStarIcon from './header-star-icon.svg'
import HeartIconGrey from './heart-icon-grey.svg'
import HeartIconWhite from './heart-icon-white.svg'
import HomeCircleIconBlue from './home-circle-icon-blue.svg'
import HomeCircleIconWhite from './home-circle-icon-white.svg'
import HomeIconGrey from './home-icon-grey.svg'
import HomeIconWhite from './home-icon-white.svg'
import InfoCircleIcon from './info-circle-icon.svg'
import ListIcon from './list-icon.svg'
import LunesIcon from './lunes-icon.svg'
import MenuIcon from './menu-icon.svg'
import OpenLockIcon from './open-lock-icon.svg'
import QRCodeIcon from './qr-code-icon.svg'
import RepeatIcon from './repeat-icon.svg'
import ShareIcon from './share-icon.svg'
import StartIconGrey from './star-icon-grey.svg'
import StartIconWhite from './star-icon-white.svg'
import TrashIcon from './trash-bin-icon.svg'
import TrophyIcon from './trophy-icon.svg'
import VolumeUpCircleIcon from './volume-up-circle-icon.svg'

export {
  AddCircleIcon,
  ArrowLeftCircleIconBlue,
  ArrowLeftCircleIconWhite,
  ArrowRightIcon,
  ArrowRightCircleIconWhite,
  BannerGreen,
  BannerRed,
  BannerYellow,
  BookIconGrey,
  BookIconWhite,
  CheckCircleIcon,
  CheckCircleIconBold,
  CheckCircleIconWhite,
  CheckCircleIconGreen,
  CheckCloseCircleIcon,
  CheckCloseCircleIconBold,
  ChevronRight,
  CloseCircleIcon,
  CloseCircleIconBold,
  CloseCircleIconWhite,
  CloseCircleIconBlue,
  CloseIcon,
  CloseIconWhite,
  CloseIconRed,
  DoubleCheckCircleIconWhite,
  DoubleCheckIcon,
  HeaderCircleIcon,
  HeaderLinesIcon,
  HeaderSquareIcon,
  HeaderStarIcon,
  HeartIconGrey,
  HeartIconWhite,
  HomeCircleIconBlue,
  HomeCircleIconWhite,
  HomeIconGrey,
  HomeIconWhite,
  InfoCircleIcon,
  ListIcon,
  LunesIcon,
  MenuIcon,
  OpenLockIcon,
  QRCodeIcon,
  RepeatIcon,
  ShareIcon,
  StartIconGrey,
  StartIconWhite,
  TrashIcon,
  TrophyIcon,
  VolumeUpCircleIcon
}
