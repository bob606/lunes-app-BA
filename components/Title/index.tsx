import {React, styles, View} from './imports';

const Title = (props) => <View style={styles.title}>{props.children}</View>;

export default Title;
