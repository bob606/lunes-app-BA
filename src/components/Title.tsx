import React, { ReactElement } from 'react'
import { View, StyleSheet } from 'react-native'

export const styles = StyleSheet.create({
  title: {
    minHeight: 54,
    alignItems: 'center',
    marginBottom: 32
  }
})

export interface ITitleProps {
  children: ReactElement
}

const Title = ({ children }: ITitleProps): ReactElement => <View style={styles.title}>{children}</View>

export default Title
