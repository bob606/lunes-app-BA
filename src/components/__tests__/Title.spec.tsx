import { render, RenderAPI } from '@testing-library/react-native'
import React from 'react'
import { Text } from 'react-native'

import { CheckCircleIcon } from '../../../assets/images'
import wrapWithTheme from '../../testing/wrapWithTheme'
import Title from '../Title'

describe('ListTitle', () => {
  const defaultTitleProps: React.ComponentProps<typeof Title> = {
    title: 'Title',
    description: 'Description'
  }

  const renderListTitle = (otherProps: Partial<React.ComponentProps<typeof Title>> = {}): RenderAPI =>
    render(<Title {...defaultTitleProps} {...otherProps} />, { wrapper: wrapWithTheme })

  it('should render ListTitle with title and description', () => {
    const { getByText } = renderListTitle()

    expect(getByText(defaultTitleProps.title)).toBeTruthy()
    expect(getByText(defaultTitleProps.description)).toBeTruthy()
  })

  it('should render ListTitle with subtitle icon and children', () => {
    const subtitle = 'Subtitle'
    const titleIcon = <CheckCircleIcon accessibilityLabel='correct' />
    const childText = 'Child'
    const { getByText, getByA11yLabel } = renderListTitle({ subtitle, titleIcon, children: <Text>{childText}</Text> })

    expect(getByText(defaultTitleProps.title)).toBeTruthy()
    expect(getByText(defaultTitleProps.description)).toBeTruthy()
    expect(getByText(subtitle)).toBeTruthy()
    expect(getByText(childText)).toBeTruthy()
    expect(getByA11yLabel('correct')).toBeTruthy()
  })
})
