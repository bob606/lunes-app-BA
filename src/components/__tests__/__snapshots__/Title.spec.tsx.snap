// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Title renders correctly across screens 1`] = `
<Styled(View)>
  <Text>
    Title
  </Text>
</Styled(View)>
`;
